torchvision/prototype/features/_image.py

```diff
@@ -122,6 +122,9 @@ def to_color_space(self, color_space: Union[str, ColorSpace], copy: bool = True)
             color_space=color_space,
         )
 
+    def to_dtype(self, dtype: torch.dtype = torch.float, copy: bool = True) -> Image:
```

Comment:
It's unclear when we are adding methods to specific `_Feature` subclasses and when we dont. In most cases, a method is added when available across all classes. This makes sense as it allows us to call these methods for every `_Feature` without checking its type. There are some limited cases (such as `BoundingBox.to_format`) where for convenience we offer specific methods to subtypes.

Why do we advocate for adding a method here? If we don't really need it, we should avoid it.

Reply:
My rule so far has been to add methods to the features if something is directly related to the attributes of a tensor. The example that we have for this are

https://github.com/pytorch/vision/blob/211563fba461062268db7cdf5a83203ed9e83e6f/torchvision/prototype/features/_bounding_box.py#L64

https://github.com/pytorch/vision/blob/211563fba461062268db7cdf5a83203ed9e83e6f/torchvision/prototype/features/_image.py#L113

(same for `features.Video`)

https://github.com/pytorch/vision/blob/211563fba461062268db7cdf5a83203ed9e83e6f/torchvision/prototype/features/_label.py#L54

I would put something that changes the tensor based on its `dtype` inside the same category, which is why I added it as method there.

> Why do we advocate for adding a method here? If we don't really need it, we should avoid it.

Unless there is a good reason for `convert_dtype` to be different from `convert_color_space` and `convert_format_bounding_box`, we should be consistent. Either have them all as methods or none of them. As explained above I'm leaning towards having them as methods, but I'm not going to oppose having them only as kernels or dispatchers.

Reply:
See #6801.